client/web/src/components/widgets/inputs/SwatchPairInput.vue

```diff
@@ -127,6 +128,18 @@ export default defineComponent({
 	mounted() {
 		this.$watch("primaryColor", this.updatePrimaryColor, { immediate: true });
 		this.$watch("secondaryColor", this.updateSecondaryColor, { immediate: true });
+
+		registerResponseHandler(ResponseType.UpdateWorkingColors, (responseData: Response) => {
+			const colorData = responseData as UpdateWorkingColors;
+			if (colorData) {
+				const { primary } = colorData;
+				const { secondary } = colorData;
+				const primaryButton = this.getRef<HTMLButtonElement>("primaryButton");
+				const secondaryButton = this.getRef<HTMLButtonElement>("secondaryButton");
+				primaryButton.style.setProperty("--swatch-color", `rgba(${primary.red}, ${primary.green}, ${primary.blue}, ${primary.alpha})`);
+				secondaryButton.style.setProperty("--swatch-color", `rgba(${secondary.red}, ${secondary.green}, ${secondary.blue}, ${secondary.alpha})`);
```

Comment:
These lines should, instead, just update the values of `this.primaryColor` and `this.secondaryColor`. I believe, because those values are watched, they should automatically update the rendered swatches.